github/repos_commits_test.go

```diff
@@ -305,99 +306,118 @@ func TestRepositoriesService_CompareCommits(t *testing.T) {
 	client, mux, _, teardown := setup()
 	defer teardown()
 
-	mux.HandleFunc("/repos/o/r/compare/b...h", func(w http.ResponseWriter, r *http.Request) {
-		testMethod(t, r, "GET")
-		fmt.Fprintf(w, `{
-		  "base_commit": {
-		    "sha": "s",
-		    "commit": {
-		      "author": { "name": "n" },
-		      "committer": { "name": "n" },
-		      "message": "m",
-		      "tree": { "sha": "t" }
-		    },
-		    "author": { "login": "l" },
-		    "committer": { "login": "l" },
-		    "parents": [ { "sha": "s" } ]
-		  },
-		  "status": "s",
-		  "ahead_by": 1,
-		  "behind_by": 2,
-		  "total_commits": 1,
-		  "commits": [
-		    {
-		      "sha": "s",
-		      "commit": { "author": { "name": "n" } },
-		      "author": { "login": "l" },
-		      "committer": { "login": "l" },
-		      "parents": [ { "sha": "s" } ]
-		    }
-		  ],
-		  "files": [ { "filename": "f" } ],
-		  "html_url":      "https://github.com/o/r/compare/b...h",
-		  "permalink_url": "https://github.com/o/r/compare/o:bbcd538c8e72b8c175046e27cc8f907076331401...o:0328041d1152db8ae77652d1618a02e57f745f17",
-		  "diff_url":      "https://github.com/o/r/compare/b...h.diff",
-		  "patch_url":     "https://github.com/o/r/compare/b...h.patch",
-		  "url":           "https://api.github.com/repos/o/r/compare/b...h"
-		}`)
-	})
+	cases := []struct {
+		name string
+		b    string
+		h    string
+	}{
+		{name: "case 1", b: "b", h: "h"},
+		{name: "case 2", b: "#b#", h: "#h#"},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			b := url.QueryEscape(c.b)
+			h := url.QueryEscape(c.h)
```

Comment:
Hi, acutally I store the escaped value for head and base in `h` and `b` and use the variables in the test instead of using literal values.